cfme/tests/infrastructure/test_quota.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+import fauxfactory
+import pytest
+from riggerlib import recursive_update
+
+from cfme.infrastructure.provider.rhevm import RHEVMProvider
+from cfme.infrastructure.provider.virtualcenter import VMwareProvider
+from cfme.markers.env_markers.provider import ONE_PER_TYPE
+from cfme.provisioning import do_vm_provisioning
+from cfme.utils.generators import random_vm_name
+from cfme.utils.update import update
+
+pytestmark = [
+    pytest.mark.provider([RHEVMProvider, VMwareProvider], scope="module", selector=ONE_PER_TYPE)
+]
+
+
+@pytest.fixture(scope='module')
+def admin_email(appliance):
+    """Required for user quota tagging services to work, as it's mandatory for it's functioning."""
+    user = appliance.collections.users
+    admin = user.instantiate(name='Administrator')
+    with update(admin):
+        admin.email = 'xyz@redhat.com'
+    yield
+    with update(admin):
+        admin.email = ''
+
+
+@pytest.fixture
+def vm_name():
+    return random_vm_name(context='quota')
+
+
+@pytest.fixture
+def template_name(provider):
+    if provider.one_of(RHEVMProvider):
+        return provider.data.templates.get('full_template')['name']
+    elif provider.one_of(VMwareProvider):
+        return provider.data.templates.get('big_template')['name']
+
+
+@pytest.fixture()
+def prov_data(vm_name):
+    return {
+        "catalog": {'vm_name': vm_name},
+        "environment": {'automatic_placement': True},
+    }
+
+
+@pytest.fixture(scope='module')
+def test_domain(appliance):
+    domain = appliance.collections.domains.create('test_{}'.format(fauxfactory.gen_alphanumeric()),
+                                                  'description_{}'.format(
+                                                      fauxfactory.gen_alphanumeric()),
+                                                  enabled=True)
+    yield domain
+    if domain.exists:
+        domain.delete()
+
+
+@pytest.fixture(scope='module')
+def max_quota_test_instance(appliance, test_domain):
+    miq = appliance.collections.domains.instantiate('ManageIQ')
+
+    original_instance = miq. \
+        namespaces.instantiate('System'). \
+        namespaces.instantiate('CommonMethods'). \
+        classes.instantiate('QuotaMethods'). \
+        instances.instantiate('quota_source')
+    original_instance.copy_to(domain=test_domain)
+
+    original_instance = miq. \
+        namespaces.instantiate('System'). \
+        namespaces.instantiate('CommonMethods'). \
+        classes.instantiate('QuotaStateMachine'). \
+        instances.instantiate('quota')
+    original_instance.copy_to(domain=test_domain)
+
+    instance = test_domain. \
+        namespaces.instantiate('System'). \
+        namespaces.instantiate('CommonMethods'). \
+        classes.instantiate('QuotaStateMachine'). \
+        instances.instantiate('quota')
+    yield instance
```

Comment:
Required: use return here, as we don't have any cleanup or further actions with instance